### ISSUE TITLE
inbox.ocaml.org VM

### ISSUE THREAD

**Issue by avsm** [ocaml/infrastructure#15]
@nojb has got everything ready (as a Dockerfile) for a inbox.ocaml.org replacement, but it needs provisioning as a Scaleway VM.  Depends on having #14 for backups first, so history doesn't repeat itself.

**Comment by avsm** [comment]
Backups are now online, so we can now look at getting Inbox back.  I'm trying hard to minimise the energy footprint of the ocaml.org infrastructure, so I'm wondering if we can do one better than having a VM dedicated to inbox.ocaml.org.

@mtelvers, might you have time to look into whether:

- we can maintain a Git version of the caml-list mbox, via some mail listening service (ideally, that we don't have to host ourselves). @nojb has one here: https://github.com/nojb/caml-list-mirror
- then, can we do a static rebuild of inbox.ocaml.org using public-inbox, and just serve the HTML, instead of having a perl service running.
- this could all be coordinated rebuilds via deploy.ocaml.org, rather than bespoke cron scripts.
- if this works, then we can also do the same with the lists.ocaml.org mbox archives and combine those into one mailing list service.
- bonus points if the resulting HTML can be styled to be compatible with the ocaml.org design in the future, so we can integrate it properly :-)

If this is all too much trouble, and we need to just get inbox.ocaml.org back in its former form, then we can do that too, but it would be nice to look into whether we can minimise our ongoing energy costs.

**Comment by avsm** [comment]
This came up again on https://github.com/ocaml/ocaml/pull/12420 -- just bumping this for @mtelvers and @tmattio to consider as part of the ocaml.org website work.

**Comment by leahneukirchen** [comment]
So is this ever coming back or not?

**Comment by dra27** [comment]
Linking in [this Discuss note from @nojb](https://discuss.ocaml.org/t/inbox-ocaml-org-down/10530/6), it was being mirrored at https://inbox.vuxu.org/caml-list/, so the archive is at least still available online up to 2022.

**Comment by leahneukirchen** [comment]
Yes, I host that archive :sweat_smile: I was wondering if there is interest in running an official version again.

**Comment by mtelvers** [comment]
Yes, definitely.  I'll try to moving this along this week.

**Comment by mtelvers** [comment]
I have made some progress.  https://inbox.ci.dev/caml-list/

**Comment by edwintorok** [comment]
> I have made some progress. https://inbox.ci.dev/caml-list/

Thanks, that archive seems to work nicely (it has recent messages from today). Took me a while to find a working web archive (the one at https://sympa.inria.fr/sympa/arc/caml-list asks for a login), should the link on OCaml.org be updated to (also) point at this one?
https://ocaml.org/community currently points to https://inbox.vuxu.org/caml-list, which although useful to have, as mentioned above is stuck at 2022.

**Comment by nojb** [comment]
@mtelvers Thanks for working on this! I had missed your original message from last year. Out of curiosity: is the archive at https://inbox.ci.dev/caml-list/ complete, or does it have gaps?

**Comment by mtelvers** [comment]
I believe https://inbox.ci.dev/caml-list/ is complete. @dra27 provided me with the missing 18 months of emails. I have a fork of @nojb's repo at https://github.com/mtelvers/caml-list/tree/master.

**Comment by leahneukirchen** [comment]
Great, then I'll make my mirror update from it again so we have another copy.
